src/queries.ts

```diff
@@ -47,43 +42,93 @@ async function getStakedPSTHolders(): Promise<StakedPSTHolders> {
 	return Object.fromEntries(stakedForAtLeastOneMonth);
 }
 
+/**
+ * Queries GQL for all ArDrive transactions within a range of blocks
+ * @param minBlock an integer representing the block from where to query the data
+ * @param maxBlock an integer representing the block until where to query the data
+ * @returns {Promise<GQLEdgeInterface[]>} the edges of the GQL result
+ */
 export async function getAllTransactionsWithin(minBlock: number, maxBlock: number): Promise<GQLEdgeInterface[]> {
 	const allEdges: GQLEdgeInterface[] = [];
+
+	const blockHeight = await getBlockHeight();
+	const trustedHeight = blockHeight - 50;
+
 	let hasNextPage = true;
+	let prevBlock = minBlock - 1;
 
 	while (hasNextPage) {
-		const query = createQuery(minBlock, maxBlock);
+		const query = createQuery(prevBlock + 1, Math.min(maxBlock, trustedHeight));
 		const response = await sendQuery(query);
-		allEdges.push(...response.edges);
-		hasNextPage = response.pageInfo.hasNextPage;
+		if (response.edges.length) {
+			console.log(`Transactions count: ${response.edges.length}`);
+			const mostRecientTransaction = response.edges[response.edges.length - 1];
+			const height = mostRecientTransaction.node.block.height;
+			writeFileSync(gqlResultName(prevBlock + 1, height), JSON.stringify(response.edges));
```

Comment:
We're writing to disk right here in the query implementation? 

Where are we consuming this?

Reply:
Yet nowhere...